src/transformers/modeling_tf_bart.py

```diff
@@ -0,0 +1,1215 @@
+            encoder_outputs = inputs[4] if len(inputs) > 4 else encoder_outputs
+            past_key_values = inputs[5] if len(inputs) > 5 else past_key_values
+            use_cache = inputs[6] if len(inputs) > 6 else use_cache
+            output_attentions = inputs[7] if len(inputs) > 7 else output_attentions
+            output_hidden_states = inputs[8] if len(inputs) > 8 else output_hidden_states
+            return_dict = inputs[9] if len(inputs) > 9 else return_dict
+        elif isinstance(inputs, (dict, BatchEncoding)):
+            assert len(inputs) <= 10, "Too many inputs."
+            if "inputs" in inputs:
+                warnings.warn("Using `inputs` as a keyword argument is deprecated. Please use `input_ids` instead.")
+            input_ids = inputs.get("input_ids")
+            attention_mask = inputs.get("attention_mask", attention_mask)
+            decoder_input_ids = inputs.get("decoder_input_ids", decoder_input_ids)
+            decoder_attention_mask = inputs.get("decoder_attention_mask", decoder_attention_mask)
+            encoder_outputs = inputs.get("encoder_outputs", encoder_outputs)
+            past_key_values = inputs.get("past_key_values", past_key_values)
+            use_cache = inputs.get("use_cache", use_cache)
+            output_attentions = inputs.get("output_attentions", output_attentions)
+            output_hidden_states = inputs.get("output_hidden_states", output_hidden_states)
+        else:
+            input_ids = inputs
+
+        use_cache = use_cache if use_cache is not None else self.config.use_cache
+        if decoder_input_ids is None:  # Classification
+            use_cache = False
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        if not use_cache:
+            decoder_input_ids, decoder_padding_mask, causal_mask = self._prepare_bart_decoder_inputs(
+                inputs,
+                decoder_input_ids=decoder_input_ids,
+                decoder_attn_mask=decoder_attention_mask,
+                mask_dtype=self.shared.dtype,
+            )
+        else:
+            decoder_padding_mask, causal_mask = None, None
+        assert decoder_input_ids is not None
+        assert isinstance(encoder_outputs, TFBaseModelOutput) or encoder_outputs is None, f"{type(encoder_outputs)}"
+        if encoder_outputs is None:
+            encoder_outputs = self.encoder(
+                input_ids=input_ids,
+                attention_mask=attention_mask,
+                output_attentions=output_attentions,
+                output_hidden_states=output_hidden_states,
+                return_dict=True,
+            )
+        assert isinstance(encoder_outputs, TFBaseModelOutput)
```

Comment:
assert messages would be nice

Reply:
Agree with Patrick, some assertion messages here would be very welcome.